server/py/services/api/crud/secrets.py

```diff
@@ -839,6 +842,33 @@ def mount_secret_token_to_runtime(
             )
         return runtime
 
+    @staticmethod
+    def _remove_auth_secret_volumes(runtime):
+        volumes = runtime.spec.volumes or []
+        mounts = runtime.spec.volume_mounts or []
+
+        volumes_to_remove = set()
+
+        # Identify volumes to remove
+        for vol in volumes:
+            if "secret" in vol:
+                secret = vol["secret"]
+                secret_name = secret.get("secretName", "")
+
+                # Pattern of auth secret volumes
+                if secret_name.startswith("mlrun-auth-secrets"):
+                    volumes_to_remove.add(vol["name"])
+
+        # Filter out only the matched volumes
+        runtime.spec.volumes = [
+            v for v in volumes if v["name"] not in volumes_to_remove
+        ]
```

Comment:
```suggestion
        runtime.spec.volumes = [
            volume
            for volume in volumes 
            if volume["name"] not in volumes_to_remove
        ]
```

use full name, makes it more readable